### ISSUE TITLE
New feature to disable specific optimizations on a per-element basis

### ISSUE THREAD

**Issue by JoyceBabu** [apache/incubator-pagespeed-mod#1320]
Is it possible to instruct ModPagespeed to skip a specific `<style>` tag? I am including critical css in the HTML when a user visits the mobile page for the first time. Pagespeed is outlining the included CSS. Is it possible to disable outlining for that specific `style` tag?

**Comment by jmarantz** [comment]
I don't think so, but I am curious what value you are getting with
css-outlining on your site.  Maybe it's easier just to turn off outlining?

-Josh

**Comment by JoyceBabu** [comment]
Thank you for the response, Joshua.

There used to be several pages with lots of inline css. I have updated most of the templates and have either removed or outlined the CSS manually. So, I think it should be fine to disable outlining altogether.

Or better still, I will disable CSS outlining on pages with critical css by setting `PageSpeedFilters: -outline_css` response header.

But I think it would be a nice feature to have the ability to disable MPS optimizations for a single element, similar to the `data-pagespeed-no-defer` attribute.

**Comment by jmarantz** [comment]
Agreed; let's leave this open for that feature then.

**Comment by jeffkaufman** [comment]
Something like `<style data-pagespeed-filters="-outline_css">...</style>` would be nice.  It could take the same syntax as `?PageSpeedFilters=...`.

**Comment by jmarantz** [comment]
data-pagespeed-filters would be powerful but maybe hard/expensive to implement in the way you might expect it to work.

For example, if you want to generally have rewrite_css off for a page, but turn it on for just one inline CSS, we wouldn't even have created a "new CssFilter" for the request.  I think it might be cumbersome to 'new' one for a particular element.

It might be easier to selectively disable stuff, say:
  data-pagespeed-disable=outline_css

We'd **still** have a probem if you try to disable rewrite_css on an external CSS link, but we then ipro-rewrite it.  You'd need to add cache-control:no-transform to the resource itself in the server, which would in fact be sufficient in that case; you wouldn't need to tweak the markup.

**Comment by jeffkaufman** [comment]
You're right; letting people disable filters for specific html elements is pretty practical, enabling filters isn't.

I think we should just document that this doesn't affect IPRO, and `no-transform` is the way to do that.  With html it's reasonable to want some control: these are ok to inline, these aren't, but with ipro pretty much you either want us to make it as small as possible in place or you want us (and anyone else) to leave it alone.
